### ISSUE TITLE
New release please

### ISSUE THREAD

**Issue by thany** [transmission-remote-gui/transgui#1428]
Is there ever going to be a new release? The latest one dates back from October 2019, which is longer ago than you might realise. I'm sure _some_ bug have been resolved. Is there any rhyme or reason as to what justifies a new release?

The current one still contains some really nasty bugs, and I can't help but to have confidence in the devs to have technically already solved them, but not (yet) made the release.

Surely after more than 2,5 years a new release is due, even if it only contains only a small number of fixes, yeah?

**Comment by xavery** [comment]
If you take a look at the diff between master and `v5.18.0`, you'll see that there's wasn't really a lot of activity since the last release. Most of the changes have either been license, translation or README updates.

The only actual feature implemented since then is support for labels introduced in #1305, and one absolute path error bugfix in #1315.

**Comment by Pentaphon** [comment]
> Is there ever going to be a new release? The latest one dates back from October 2019, which is longer ago than you might realise. I'm sure _some_ bugs have been resolved. Is there any rhyme or reason as to what justifies a new release?
> 
> The current one still contains some really nasty bugs, and I can't help but to have confidence in the devs to have technically already solved them, but not (yet) made the release.
> 
> Surely after more than 2,5 years a new release is due, even if it only contains only a small number of fixes, yeah?

You can try 
https://github.com/equeim/tremotesf2 
 https://github.com/transmission-remote-gtk/transmission-remote-gtk
https://github.com/Rebell81/Kebler
 in the meantime.

**Comment by thany** [comment]
Thanks! Tremotesf2 feel a lot more mature in behaviour. It doesn't freeze at the drop of a hat (or indeed a magnet link), it doesn't freeze every time the server doesn't respond for 0.2 femtoseconds. And the GUI scales properly, which is something I just simply expect in 2019, nevermind 2022.

**Comment by qu1ck** [comment]
Transgui has been very unstable for me, crashing a lot and freezing. Also I can't stand pascal so can't be arsed to improve the code.

Instead I've been working on a rewrite of transgui with rust backend and react frontend built with tauri. It's crossplatform like electron but doesn't ship whole browser so will have comparable binary size to transgui (<10mb).

![image](https://user-images.githubusercontent.com/3622532/184512018-8992014c-a746-45a1-9047-706cb162e78e.png)

It looks like shit right now but getting there in terms of functionality.

**Comment by thany** [comment]
@qu1ck Don't try too hard to emulate whatever Transgui does, and don't try too hard to mimick a native application. Best do what comes natural in an Electron app, otherwise there be dragons. Assuming you are a solo or small team, you won't have the development capacity of, say, VScode 😀  They can go all out supporting every edge-case they come across.

Try to make something you feel comfortable with - this may not be 100% identical to Transgui, but something that "feels like it".

**Comment by Pentaphon** [comment]
>It's crossplatform like electron but doesn't ship whole browser so will have comparable binary size to transgui (<10mb).

Doesn't seem as exciting as a native app, IMHO.

**Comment by qu1ck** [comment]
@thany oh I'm already deviating from Transgui where it makes sense to me (see the tab bar and directory tree structure). But I like the general layout of the app and it makes for a great starting point. VSCode is several orders of magnitude larger project, this is definitely something a solo dev can handle.

@Pentaphon it's a good thing that I'm not going for exciting then :)

**Comment by thany** [comment]
> > It's crossplatform like electron but doesn't ship whole browser so will have comparable binary size to transgui (<10mb).
> 
> Doesn't seem as exciting as a native app, IMHO.

Curious to now what would be considered a native app? Perhaps an Electron app is right on the cusp - it probably depends on who you ask 🙂

**Comment by qu1ck** [comment]
There are 2 more or less defined thresholds between a web app and a native app.
1. Full accessibility of system APIs
2. Use of native GUI widgets (i.e. buttons, textboxes, radio switches, tabs and other controls)

First is about the app working as a native app, second is about it looking like one.

Electron has neither. It exposes a limited subset of system APIs and you can't use native GUI widgets.

Tauri gives you full access to system APIs from it's backend so your application is not limited. GUI is still just a web gui. It's a detriment if you want your app to look native but a benefit if you want easy cross platform UI. It's a double benefit if you also plan to use your app as web app, which I do. Transmission supports custom web interfaces and I can easily repackage my app to be a custom web skin for transmission daemon. You won't need to have a native app at all but it will also lose some functionality because of browser limitations.

**Comment by Mrnofish** [comment]
FWIW this client was modeled after uTorrent, at least in the form it was back at the time - haven't used that program in a long time, tho, and I cannot say how much the two might have deviated.

As for the opportunity of having clients running on Electron, IMO, for the end user, that begs the question: why would I rather not go server-side instead.

That shouldn't stop anyone having fun building stuff, however in terms of how many users might be interested in using a client-side Electron app VS a server-side web UI, that's a legit questions as I'm struggling to see the benefits of the former over the latter.

**Comment by qu1ck** [comment]
For me there is one killer feature that is impossible on server side: local path mappings.
You can configure server's `/mnt/data/torrents` to be interpreted as `/mnt/seedboxnfs` or `Z:\torrents` on your local system and then open files directly from transgui.

That alone is worth it to me to put effort into native-ish app.

There is also system notifications for completed torrents but that's just icing on top.

**Comment by Mrnofish** [comment]
That's a valid point for sure, though I personally consider path mapping more of a goodie than an essential one. In a Windows scenario, folder shortcuts and Quick Access pins provide reasonably convenient ways to navigate to a download folders, although maybe not as fast and practical as simply double clicking on a torrent.

**Comment by username227** [comment]
I cannot get Transgui path mapping to work on any Cinnamon distro except for Linux Mint; i've dropped entire distro experiments because of the path mapping issue.
Regarding people's point about utorrent, I also haven't used it in years, but I find the interface of utorrent/qbittorrent and by extension transgui, much more intuitive than transmission. I sincerely hope that somebody will make a native app that is less buggy and keeps a similar appearance. There's a reason that we're here in this thread having this conversation about a program that hasn't put out a new version in two years - the UI is decent, and the second best is far behind.

**Comment by Gertcha** [comment]
There is a native macOS app of Transmission 4.0.0-beta.1 out, do you think that will have the ability to access remote instances?

**Comment by scineram** [comment]
Now that Transmission 4 is official, which I haven't updated yet, is this still expected to work? Or should I look for a different client?

**Comment by xavery** [comment]
tl;dr yes

The only breaking change in 4.0 in terms of RPC is deprecating the undocumented `/upload` endpoint which transgui has never used anyway.

`session-get.download-dir-free-space` has been deprecated but that doesn't affect transgui as it even now uses the recommended `free-space` method.

`torrent-set.tracker{Add,Remove,Replace}` have been deprecated by `trackerList` which will be of concern in future versions.

**Comment by qu1ck** [comment]
If people are still looking for a similar app: I just published first version of TrguiNG (almost a year since I posted first screenshots in a post above)

https://github.com/openscopeproject/TrguiNG

Builds for all platforms are available in release page. You can eve use it as a web interface served by transmission itself, infinitely more useful than the original web interface and still has 90% of functionality of native app.

I consider it to be in beta: stable, functional and usable but may need some polish here and there. Please report any issues you may find if you try this app.

**Comment by Mrnofish** [comment]
@qu1ck Not exactly thrilled that the setup tries to install Microsoft Edge WebView2, TBH. Is there no alternative to install the native app on Windows ?

**Comment by qu1ck** [comment]
No, it needs the webview. It is preinstalled on modern windows anyway, you either have win 7 which is not supported or very old build of win 10 which you should update.

**Comment by Mrnofish** [comment]
> It is preinstalled on modern windows anyway, you either have win 7 which is not supported or very old build of win 10 which you should update.

I am on 19045.3086 which is the June cumulative for 22H2. Better luck next time!

**Comment by qu1ck** [comment]
My mistake, I guess it's only auto installed on managed win 10 installations for now https://blogs.windows.com/msedgedev/2022/12/14/delivering-microsoft-edge-webview2-runtime-to-managed-windows-10-devices/

Either way, there is no reason to avoid it, you have edge already, webview is just a library that allows using edge's engine in other apps. TrguiNG uses it for displaying it's UI so there is no way around it.

**Comment by Mrnofish** [comment]
It's true they force stuff on you like they, not you, owned the metal, however I have Edge firewalled until the next PC, and then I'll be switching to Linux on the desktop, too, as I'm more than fed up with having to constantly check what these spooks are trying to do behind my back.

I don't suppose the Mac and Linux ports rely on Edge, though?

**Comment by qu1ck** [comment]
No, mac and linux builds use native webkit web view.

**Comment by thany** [comment]
I'm using [tremotesf](https://github.com/equeim/tremotesf2). It works pretty well and doesn't require the co-installation of an almost entire web browser.

**Comment by qu1ck** [comment]
tremotesf looks nice but it's simplistic, almost like native transmission clients. It looks like a good choice if you don't need label filtering, concurrent connections, torrent file renaming, torrent details at a glance and other features.

> doesn't require the co-installation of an almost entire web browser

Neither does TrguiNG. Webview is a small library that allows using part of the browser in other apps, it's not a browser. And the whole topic is only relevant for windows 10 which does't have webview preinstalled.

**Comment by fredo-47** [comment]
For me on Linux, TrguiNG took an eternity to build.
Unlike transgui or xavery's fork of transgui

**Comment by qu1ck** [comment]
Rust is famous for slow linking but even underpowered github runners can do the job in ~10 minutes. It's only an issue for first compile though.

**Comment by Pentaphon** [comment]
> I'm using [tremotesf](https://github.com/equeim/tremotesf2). It works pretty well and doesn't require the co-installation of an almost entire web browser.

I also have settled on using tremotesf. It has the most promise and an active dev. The only thing I would add is a 3 pane UI similar to transgui.

**Comment by Mrnofish** [comment]
> And the whole topic is only relevant for windows 10 which does't have webview preinstalled.

Unfortunately, Windows is still the dominant desktop OS.

Mind, I'm happy that there is another alternative in the ecosystem and yours seems a nifty client... perhaps accept that some people might not be super cool with certain choices ?

It's just a case of you can't make everybody happy: nobody is passing judgment or asking that you do different.

I also concur on tremotesf being a bit too barebones FWIW, but different strokes for different folks. There does not need to be The One Client to Rule Them All.

**Comment by qu1ck** [comment]
I am perfectly fine with your choice of client, whatever works for you is great. I don't force my client on anyone.

I'm just correcting the notion that webview dependency is like having to download another browser. It's not. In fact, it's the reason this app uses tauri and not electron like discord/vscode/other apps. I didn't want to ship a browser with my app.

> Unfortunately, Windows is still the dominant desktop OS.

And yes, windows is the dominant os. OS which already forces edge install on you. Having a small library that makes that bloatware at least somewhat useful to non-edge users is a small price to pay, in my opinion.

I specified windows 10, which while still dominant, won't be dominant forever and windows 11 already has webview preinstalled.

**Comment by Mrnofish** [comment]
It being pre-installed in some instances is a moot point, as long as there are ways to prevent it from connecting to the local network and/or Internet, possibly from running altogether, and perhaps it can be even removed from the system or may be in the future.

Anyway, if your project relied on a piece of FOSS software to run the web side of things, none of this conversation would've even sprung up.

It's not that FOSS is a panacea, but rather that Microsoft has been so hell-bent on making people dance to their tune, so brazen (see e.g. the Windows 11 hardware requirement debacle, even the timing was an epic fail), that trust in the company may be close to a historical low (which is indeed a feat for a company with such a checkered history) and some types of users will try to have as little Microsoft on the Windows systems that cannot be realistically switched to another OS.

**Comment by qu1ck** [comment]
Like I said, blocking edge has no effect on TrguiNG, it doesn't use edge executable. Somehow uninstalling/ripping it out is likely to break your system. There's a reason it is not optional, other system things depend on it same as IE used to be integral. I don't think there will be a way to remove it in the future, at least not until it is replaced with another system integrated browser.

If there was a FOSS option for windows I would consider it. As far as I know there isn't one.

Microsoft has done a lot of stupid things, some of which was forcing users to do things they didn't want to do for no good reason and some of which was failing to explain why they are forcing some things that have good reasons. Microsoft is also way bigger than windows and some media noise about tpm requirements will not change anything, sadly.

> some types of users will try to have as little Microsoft on the Windows systems that cannot be realistically switched to another OS.

I understand the sentiment but these efforts are mostly pointless. Sure, disable the telemetry service, you still can never be sure what some other part you can't disable is doing. Fundamentally the only choice that can give some peace of mind is open source, and even that is not a guarantee. 

To summarize this discussion, there is plenty of choice:
* Use TrguiNG as native app with webview on windows
* Use TrguiNG as native app on non-windows OS
* Use TrguiNG as web interface with browser of your choice
* Use any other client or web interface that suits your needs better

I understand that the weirdly narrow "Use TrguiNG as native app but without microsoft stuff but still on microsoft os" option is missing. I'm fine with that. Like you said:

> No need for The One Client to Rule Them All.

**Comment by antekgla** [comment]
[qu1ck ](https://github.com/qu1ck) Any reason for no provide **TrguiNG** a portable version for Windows?
My Chrome browser blocks the setup exe download in the release page marking it as dangerous and VirusTotal flags it danger in 3 vendors.
I know what false positives are very common but given this flags I don't want to grant Administrator rights to this file.

I really want to try it because looks VERY VERY good...

**Comment by qu1ck** [comment]
@antekgla The installer is necessary to ensure you have webview installed and to create correct shortcuts. That's why there will not be a portable executable.

Here's a few things you can do:
* Try the .msi installer, it's exactly same as the other one, just uses older windows distribution method.
* Use 7-zip or similar to extract .exe directly from the installer archive. You will have to make sure you have webview installed if you have windows 10. The app will work but the system notifications for completed torrents may not show the correct icon and program name (that's what "correct" shortcuts are for).
* Compile it yourself. You don't need to be a developer, it's as simple as installing node, rust and running 2 commands in terminal.

**Comment by Mrnofish** [comment]
@qu1ck you are coming across as oddly cavalier about this matter.

While you are entitled to your opinion, statements like:

> these efforts are mostly pointless

are not going to buy you any love or respect. It's not just that having an opinion should never cross into entitlement territory.

Crucially, the bulk of this kind of arguments have been debunked, so readers are more and more inclined to chalk it up not to ignorance, but to intellectual dishonesty.

**Comment by qu1ck** [comment]
I have no idea where you see entitlement in my statements. Cavalier - sure, I guess, if you squint hard enough.

I stand by my opinion that trying to "fix" windows by unsupported commands, registry edits and other manual interventions is at best pointless overall. I don't dispute the validity of privacy concerns, I am pointing out that it's like trying to board up your (pun not intended) bathroom window while willingly giving a copy of a key to your main entrance door to someone you don't trust to be an entirely upstanding citizen. 

I understand that some people are stuck with windows and have to deal with it. I'm not sure there are entirely effective ways to deal with it but uninstalling edge is one of the most ineffective. Block it with firewall? Sure. Quarantine the binary? Questionable but it's probably fine. Uninstall it completely? Be prepared that one day your windows will start taking 10 seconds to show start menu or an update will fail to install or a number of other seemingly unrelated things break. Just because you didn't immediately get a BSOD doesn't mean your system works as intended.

> Crucially, the bulk of this kind of arguments have been debunked

Depending on how broad your definition of "bulk" and "this kind" are, the arguments range from "debunked" all the way to "trivially verifiable to be true". 

I don't see why we are continuing this discussion here. This has nothing to do with transmission or it's clients. Feel free to email me if you want to continue, I will only answer to TrguiNG related posts.

**Comment by Mrnofish** [comment]
> This has nothing to do with transmission or it's clients

It's about somebody who's promoting their software on a different project's page and in spite of all the goodwill shown to them, they still can't get a hold of themselves, with a range of bossy, arrogant antics and an interested ideology.

Great job with the promotion, BTW, I'll make a point to avoid anything you put hands on, because why would anybody trust somebody who acts and talks that way?

Now be a chum, pick up your backseat moderation and the soapbox attitude and take it where it belongs.

**Comment by garoto** [comment]
I already unstarred it and removed all traces of "EdgeWebView" from my (8.1) system. Plus, it didn't even worked with my local machine `ssh` forwarder.

**Comment by Tharn** [comment]
Edge is not "integrated" into the OS the same way that Internet Explorer was with older versions of Windows. What the OS does is invoke Edge when it wants to show you a link from the Control Panel for example. It's removable **without issues** and has nothing to do with the internals of the OS.

```
cd %Program Files (x86)%\Microsoft\Edge\Application\EdgeVersion\Installer
setup --uninstall --force-uninstall --system-level
```

is not an unsupported command, it's literally the opposite. Microsoft puts up hurdles because they don't **like** letting people uninstall it, that's all.

**Comment by garoto** [comment]
xavery's "fork" of "trangui" is the way to go for me. His compiles have gzip support, which, like I mentioned in another related issue tracker, makes his transgui compiles to be noticeable mkore responsive with high latency links. Not to mention being compiled by a newer compiler and related libs. It's win-win.

**Comment by Pentaphon** [comment]
> xavery's "fork" of "transgui" is the way to go for me.

I like it as well but I wish all these forks would just put all their efforts together into an actively maintained community version that releases at least every 6 months or so.

**Comment by garoto** [comment]
My wishes as well, but that ain't gonna happen. It rarely does.

**Comment by qu1ck** [comment]
@Mrnofish ok, now you are just openly trolling.

@garoto if you want to describe your issue with forwarding I can try to replicate it and fix it but it's best to do it on TrguiNG github.

@Tharn Did you find that command on microsoft support or docs website? Because if not then it's literally unsupported. And you have to follow up that command with a registry edit to keep edge from being reinstalled. Will you say that's also supported?
Yes, it's not integrated same way IE used to be. At least not yet, not on windows 10. No, you can't definitively say it's removable without issues. Maybe it is, maybe it is now but will break with next update, maybe it is already broken but you will not notice until specific action is performed.

And I guess I have to repeat myself again even though I thought this was obvious: do whatever you want with your PCs, I'm not forcing anyone to use or not use this or that program. Don't like edge or webview? Cool. Don't want to install programs that have webview dependency? Your choice. 

I wrote TrguiNG because existing options did not fit my needs, I made an effort to make it work on all major operating systems and as a web interface to make it usable for majority of people and shared it as open source program. Not everyone will like it or use it and that is fine. Some people will and some people will even help make it better and that's why I published it.

@Pentaphon I mentioned this above but transgui is written in a pretty much dead language and is hard to work on. One of my pull requests was left unreviewed for 1.5 years. I gave up and started my project, it's a complete rewrite, not a fork.

**Comment by Pentaphon** [comment]
> @Mrnofish ok, now you are just openly trolling.
> 
> @garoto if you want to describe your issue with forwarding I can try to replicate it and fix it but it's best to do it on TrguiNG github.
> 
> @Tharn Did you find that command on microsoft support or docs website? Because if not then it's literally unsupported. And you have to follow up that command with a registry edit to keep edge from being reinstalled. Will you say that's also supported? Yes, it's not integrated same way IE used to be. At least not yet, not on windows 10. No, you can't definitively say it's removable without issues. Maybe it is, maybe it isn't now but will break with next update, maybe it is already broken but you will not notice until specific action is performed.
> 
> And I guess I have to repeat myself again even though I thought this was obvious: do whatever you want with your PCs, I'm not forcing anyone to use or not use this or that program. Don't like edge or webview? Cool. Don't want to install programs that have webview dependency? Your choice.
> 
> I wrote TrguiNG because existing options did not fit my needs, I made an effort to make it work on all major operating systems and as a web interface to make it usable for majority of people and shared it as open source program. Not everyone will like it or use it and that is fine. Some people will and some people will even help make it better and that's why I published it.
> 
> @Pentaphon I mentioned this above but transgui is written in a pretty much dead language and is hard to work on. One of my pull requests was left unreviewed for 1.5 years. I gave up and started my project, it's a complete rewrite, not a fork.

In that case, yeah, we should abandon it in favor of your project, tremotesf and any other project that is written in a language people can actually contribute to. No point begging the developers for a new release. I'm fine with everybody flocking to other projects. The only reason people still keep coming back to this project is because its on Github and everybody already has a github account. If transgui were stuck on sourceforge, it would have been long dead.

**Comment by Tharn** [comment]
It's still the most feature-complete native client. With two downsides. It crashes occasionally and it gobbles up a lot of bandwidth with the window open and a lot of active columns displaying data. I don't think it makes use of the transport compression that Transmission is now able to do? At any rate, tremotesf2 looks very close to being "the one" for the future. The dev is responsive, but it's a solo project also. Maybe he will accept help.

**Comment by enchained** [comment]
> it gobbles up a lot of bandwidth with the window open and a lot of active columns displaying data. I don't think it makes use of the transport compression that Transmission is now able to do?

You can try xavery fork: https://github.com/transmission-remote-gui/transgui/issues/1422
